driver/src/test/functional/com/mongodb/client/CrudTest.java

```diff
@@ -0,0 +1,271 @@
+        CountOptions options = new CountOptions();
+        if (arguments.containsKey("skip")) {
+            options.skip(arguments.getNumber("skip").intValue());
+        }
+        if (arguments.containsKey("limit")) {
+            options.limit(arguments.getNumber("limit").intValue());
+        }
+
+        return toResult((int) collection.count(arguments.getDocument("filter"), options));
+    }
+
+    private BsonDocument getDistinctResult(final BsonDocument arguments) {
+        return toResult(collection.distinct(arguments.getString("fieldName").getValue(), BsonInt32.class)
+                .filter(arguments.getDocument("filter")).into(new BsonArray()));
+    }
+
+    private BsonDocument getFindResult(final BsonDocument arguments) {
+        FindIterable<BsonDocument> findIterable = collection.find(arguments.getDocument("filter"));
+        if (arguments.containsKey("skip")) {
+            findIterable.skip(arguments.getNumber("skip").intValue());
+        }
+        if (arguments.containsKey("limit")) {
+            findIterable.limit(arguments.getNumber("limit").intValue());
+        }
+        if (arguments.containsKey("batchSize")) {
+            findIterable.batchSize(arguments.getNumber("batchSize").intValue());
+        }
+        return toResult(findIterable);
+    }
+
+    private BsonDocument getDeleteManyResult(final BsonDocument arguments) {
+        return toResult("deletedCount",
+                new BsonInt32((int) collection.deleteMany(arguments.getDocument("filter")).getDeletedCount()));
+    }
+
+    private BsonDocument getDeleteOneResult(final BsonDocument arguments) {
+        return toResult("deletedCount", new BsonInt32((int) collection.deleteOne(arguments.getDocument("filter")).getDeletedCount()));
+    }
+
+    private BsonDocument getFindOneAndDeleteResult(final BsonDocument arguments) {
+        FindOneAndDeleteOptions options = new FindOneAndDeleteOptions();
+        if (arguments.containsKey("projection")) {
+            options.projection(arguments.getDocument("projection"));
+        }
+        if (arguments.containsKey("sort")) {
+            options.sort(arguments.getDocument("sort"));
+        }
+        return toResult(collection.findOneAndDelete(arguments.getDocument("filter"), options));
+    }
+
+    private BsonDocument getFindOneAndReplaceResult(final BsonDocument arguments) {
+        FindOneAndReplaceOptions options = new FindOneAndReplaceOptions();
+        if (arguments.containsKey("projection")) {
+            options.projection(arguments.getDocument("projection"));
+        }
+        if (arguments.containsKey("sort")) {
+            options.sort(arguments.getDocument("sort"));
+        }
+        if (arguments.containsKey("upsert")) {
+            options.upsert(arguments.getBoolean("upsert").getValue());
+        }
+        if (arguments.containsKey("returnDocument")) {
+            options.returnOriginal(!arguments.getString("returnDocument").getValue().equals("After"));
```

Comment:
We use `returnOriginal` in options opposed to the spec example of a `returnDocument` enum